search-impl/src/test/java/com/example/core/TopicStub.java

```diff
@@ -49,13 +54,21 @@ public TopicStub(Materializer materializer) {
                         .via(flow)
                         .toMat(Sink.ignore(), Keep.both())
                         .run(materializer);
-                sourceActor = pair.first();
+                sourceActor.complete(pair.first());
                 return pair.second();
             }
         };
     }
 
-    public Supplier<ActorRef> actorSupplier() {
-        return () -> sourceActor;
+    /**
+     * @return a {@link ProducerStub} that clients (test code) can use to send messages through this {@link TopicStub}.
+     */
+    public ProducerStub<T> producerStub() {
+        return (T message) ->
+                sourceActor.thenApply(topicActor -> {
```

Comment:
Yeah, I like that idea.

If we want to support sending messages before there are subscribers and new subscribers seeing existing messages, we could pair that actor with a list of sent messages. Then, the `send` method could both send to the actor and add to the list. The subscribe method could concatenate a source built from the list with the actor source.